app/src/gecko/java/org/mozilla/focus/web/GeckoViewPrompt.java

```diff
@@ -0,0 +1,888 @@
+                @Override
+                public void onItemClick(final AdapterView<?> parent, final View v,
+                                        final int position, final long id) {
+                    final GeckoBundle item = adapter.getItem(position);
+                    if (type == CHOICE_TYPE_MENU) {
+                        final GeckoBundle[] children = item.getBundleArray(ITEMS);
+                        if (children != null) {
+                            // Show sub-menu.
+                            dialog.setOnDismissListener(null);
+                            dialog.dismiss();
+                            promptForChoice(session, item.getString(LABEL), /* msg */ null,
+                                    type, children, callback);
+                            return;
+                        }
+                    }
+                    callback.confirm(item);
+                    dialog.dismiss();
+                }
+            });
+        } else if (type == CHOICE_TYPE_MULTIPLE) {
+            list.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+                @Override
+                public void onItemClick(final AdapterView<?> parent, final View v,
+                                        final int position, final long id) {
+                    final GeckoBundle item = adapter.getItem(position);
+                    item.putBoolean(SELECTED, ((CheckedTextView) v).isChecked());
+                }
+            });
+            builder.setNegativeButton(android.R.string.cancel, /* listener */ null)
+                    .setPositiveButton(android.R.string.ok,
+                            new DialogInterface.OnClickListener() {
+                                @Override
+                                public void onClick(final DialogInterface dialog,
+                                                    final int which) {
+                                    final int len = adapter.getCount();
+                                    ArrayList<String> items = new ArrayList<>(len);
+                                    for (int i = 0; i < len; i++) {
+                                        final GeckoBundle item = adapter.getItem(i);
+                                        if (item.getBoolean(SELECTED)) {
+                                            items.add(item.getString("id"));
+                                        }
+                                    }
+                                    callback.confirm(items.toArray(new String[items.size()]));
+                                }
+                            });
+            dialog = createStandardDialog(builder, callback);
+        } else {
+            throw new UnsupportedOperationException();
+        }
+        dialog.show();
+    }
+
+    private static int parseColor(final String value, final int def) {
+        try {
+            return Color.parseColor(value);
+        } catch (final IllegalArgumentException e) {
+            return def;
```

Comment:
\[PMD\] INFO: The method 'promptForColor' has a Modified Cyclomatic Complexity of 10\.